src/toolkits/drawing_classifier/drawing_classifier.hpp

```diff
@@ -0,0 +1,250 @@
+/* Copyright © 2019 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+#ifndef TURI_DRAWING_CLASSIFIER_H_
+#define TURI_DRAWING_CLASSIFIER_H_
+
+#include <core/logging/table_printer/table_printer.hpp>
+#include <model_server/lib/extensions/ml_model.hpp>
+#include <core/data/sframe/gl_sframe.hpp>
+#include <toolkits/drawing_classifier/dc_data_iterator.hpp>
+#include <toolkits/coreml_export/mlmodel_wrapper.hpp>
+#include <ml/neural_net/compute_context.hpp>
+#include <ml/neural_net/model_backend.hpp>
+#include <ml/neural_net/model_spec.hpp>
+
+namespace turi {
+namespace drawing_classifier {
+
+class EXPORT drawing_classifier: public ml_model_base {
+
+ public:
+  
+  // ml_model_base interface
+
+  void init_options(const std::map<std::string, flexible_type>& opts) override;
+  size_t get_version() const override;
+  void save_impl(oarchive& oarc) const override;
+  void load_version(iarchive& iarc, size_t version) override;
+
+  // Interface exposed via Unity server
+
+  void train(gl_sframe data,
+             std::string target_column_name, std::string feature_column_name,
+             variant_type validation_data,
+             std::map<std::string, flexible_type> opts);
+  gl_sarray predict(gl_sframe data, std::string output_type);
+  gl_sframe predict_topk(gl_sframe data, std::string output_type,
+    size_t k, size_t batch_size);
+  variant_map_type evaluate(gl_sframe data, std::string metric,
+    size_t batch_size);
+  std::shared_ptr<coreml::MLModelWrapper> export_to_coreml(
+      std::string filename);
+  void load_pretrained_model(variant_map_type model_data, size_t version);
```

Comment:
What is this for? Why is loading a pre-trained model part of the public API, outside of the options for `train`?